javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -632,7 +660,12 @@ protected void generatePagedAsyncSinglePage(ClientMethod clientMethod, JavaType
             ApplyParameterTransformations(function, clientMethod, settings);
             ConvertClientTypesToWireTypes(function, clientMethod, restAPIMethod.getParameters(), clientMethod.getClientReference(), settings);
 
-            String serviceMethodCall = checkAndReplaceParamNameCollision(clientMethod, restAPIMethod, settings);
+            boolean requestOptionsLocal = false;
+            if (settings.isLowLevelClient()) {
```

Comment:
Reason is implementation in HLC would be (almost totally) different.

Quick capture, it will result in ProxyMethod have this 2 parameters, and ClientMethod filling them similar to what we done here.